### PR TITLE
[Core] Fix condition creation in sub model part

### PR DESCRIPTION
**Pull request by ddiezrod**
**📝 Description**
I have encountered a problem when trying to create a new condition with a vector of nodes. The current implementation assumes that the nodes belong to the model part, but this is not necessarily the case if it is a submodel part. For some reason, this was fixed for the elements but not for the conditions. I think it makes sense to do the same for both functions (see https://github.com/KratosMultiphysics/Kratos/blob/f6e7a3c2c529a6a8c9e782d628fe412633771220/kratos/sources/model_part.cpp#L980 )

### CODE REVIEW

File: kratos/sources/model_part.cpp
Comment (loumalouomega) [inline kratos/sources/model_part.cpp:1518]:
```suggestion
    
    if (IsSubModelPart()) {
```

**Review by loumalouomega — APPROVED** [review]
Looks Okay, but maybe wait for the approval of someone from @KratosMultiphysics/technical-committee

**Comment by ddiezrod** [comment]
Just to be clear, the problem arises when we are in line 
https://github.com/KratosMultiphysics/Kratos/blob/f6e7a3c2c529a6a8c9e782d628fe412633771220/kratos/sources/model_part.cpp#L1521

and we try to do a pGetNode. If it is a submodelpart, it might not have the node.

**Review by rubenzorrilla — APPROVED** [review]
Makes very much sense. Thanks for the fix.

**Pull request merged**
